pysat/_instrument.py

```diff
@@ -3023,20 +3027,23 @@ def remote_file_list(self, start=None, stop=None):
         return a subset of available files.
 
         """
-        # Set the user-supplied kwargs
-        if 'list_remote_files' in self.kwargs.keys():
-            kwargs = self.kwargs['list_remote_files']
-        else:
-            kwargs = {}
 
         # Add the function kwargs
         kwargs["start"] = start
         kwargs["stop"] = stop
 
+        # Add the user-supplied kwargs
+        rtn_key = 'list_remote_files'
+        if rtn_key in self.kwargs.keys():
+            for user_key in self.kwargs[rtn_key].keys():
+                # Don't overwrite kwargs supplied directly to this routine
+                if user_key not in kwargs.keys():
+                    kwargs[user_key] = self.kwargs[rtn_key][user_key]
```

Comment:
@jklenzing I decided to prioritize kwargs supplied directly to the routine over those supplied to the Instrument.